### DISCUSSION TITLE
It would be nice if renaming nodes immediately after adding them using keypresses only was more streamlined

### DISCUSSION THREAD

**Discussion by ThatBebx** [godotengine/godot-proposals#4171] [Editor]
To expand on the title - Say I'm adding a node. The process for this is: Click on the parent node -> CTRL + A -> type in the node you wish to add -> enter -> enter again to change its name -> if renaming fails to work, make sure your mouse is in the scene tree to ensure hitting enter renames -> rename.

It's that second last If step that is the issue here. It would be nice if this wasn't a question, and that whenever you added a node, it would immediately highlight that node, so that hitting enter would **always** rename that node after adding it. If the user moves the mouse after adding the node, the node should be no longer highlighted, and enter no longer renames the node.

Alternatively, though I imagine this is not the preferred solution, an unobtrusive dialogue box could pop up, where the user could rename the node, and hit enter (or hit enter straight away to leave the default name of the node). If the user clicks out of this box, the box closes with the node name remaining unchanged. (This is similar behavior to parameter boxes in Blender, like when deleting an object for example.)

A third solution is to allow renaming the node before adding it. The downside with this option is that this would require the additional keypress of pressing tab as well as enter to rename nodes fluidly, which is a bit of an interruption when compared to the other two options.

I think this would be very helpful to the flow of godot - the only reasons I can see for not adding this is that it is too low priority to be considered an issue (which, fair), or that this is contradictory to the design language of godot. Solution 3 is one which solves the problem without compromise on design language, but solution 1 I think is a reasonable compromise on previous design choices on the program.

Thanks for reading! I've nothing more to say on this.

**Comment by Mickeon** [comment]
> A third solution is to allow renaming the node before adding it. The downside with this option is that this would require the additional keypress of pressing tab as well as enter to rename nodes fluidly, which is a bit of an interruption when compared to the other two options.

For a while, I had in the back of my head the idea of a proposal to allow defining default, per-project Node names as an user-friendly text-box in the Create New Node window. I have been unable to properly articulate the reasoning to add this, however.

**Comment by theraot** [comment]
> that whenever you added a node, it would immediately highlight that node, so that hitting enter would always rename that node after adding it.

Yes.

> If the user moves the mouse after adding the node, the node should be no longer highlighted, and enter no longer renames the node.

No. Just select the Node.

---

Something similar happens when adding a file in the FileSystem (e.g. New Script), it should select it. No questions asked (Edit: I mean, it should not matter if the mouse moved or whatever).
